core-ktx/src/main/java/com/amplifyframework/kotlin/DataStore.kt

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.kotlin
+
+import com.amplifyframework.core.model.Model
+import com.amplifyframework.core.model.query.QueryOptions
+import com.amplifyframework.core.model.query.Where
+import com.amplifyframework.core.model.query.predicate.QueryPredicate
+import com.amplifyframework.core.model.query.predicate.QueryPredicates
+import com.amplifyframework.datastore.DataStoreException
+import com.amplifyframework.datastore.DataStoreItemChange
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.Flow
+
+/**
+ * A local object store with cloud synchronization.
+ */
+interface DataStore {
+    /**
+     * Save an item into the DataStore.
+     * @param item Item to save
+     * @param predicate Conditions that must be true before save can succeed.
+     *                  If not specified, an "allow all" predicate is used.
+     */
+    @Throws(DataStoreException::class)
+    suspend fun <T : Model> save(item: T, predicate: QueryPredicate = QueryPredicates.all())
+
+    /**
+     * Delete an item from the DataStore.
+     * @param item Item to delete
+     * @param predicate Conditions that must be true before delete can succeed.
+     *                  If not specified, an "allow all" predicate is used.
+     */
+    @Throws(DataStoreException::class)
+    suspend fun <T : Model> delete(item: T, predicate: QueryPredicate = QueryPredicates.all())
+
```

Comment:
Yes! Good catch. Will add now.

Reply:
On closer inspection, that new `delete` API in the Java API surface isn't ready for prime time. It can delete many things, but it has a return type for a single deletion. I'm inclined to not include it in the Kotlin API surface right now, since it will need some rework. https://github.com/aws-amplify/amplify-android/pull/1106